test/controllers/tasks_controller_test.rb

```diff
@@ -123,5 +123,29 @@ class TasksControllerTest < ActionController::TestCase
       assert_response :success
       assert_json_keys %w(status output rollback_url)
     end
+
+    test ":lookup returns stack task url if the task is an instance of Task" do
+      @task = shipit_tasks(:shipit_restart)
+
+      get :lookup, params: {id: @task.id}
+
+      assert_redirected_to stack_task_path(@task.stack, @task)
+    end
+
+    test ":lookup returns stack deploy url if the task is an instance of Deploy" do
+      @task = shipit_tasks(:shipit)
+
+      get :lookup, params: {id: @task.id}
+
+      assert_redirected_to stack_deploy_path(@task.stack, @task)
+    end
+
+    test ":lookup returns stack deploy url if the task is a kind of Deploy" do
```

Comment:
nit `kind of Rollback`?

Reply:
actually not, the business logic I want to test is: given the instance from a subclass of `Deploy` 
it should redirect to the deploy resource url, in this example I'm providing a `Rollback` instance.

I can change to `:lookup returns stack deploy url if the task is an instance of a subclass of Deploy`, what do you think?

Reply:
From the test perspective, we should forget that any subclassing exists. This test is explicitly testing that Rollbacks receive a deploy url, so we should probably label it like that. It's a single case we're testing here, not a generalization 🙂